hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/config/OAuthConfiguration.java

```diff
@@ -0,0 +1,90 @@
+import pl.allegro.tech.hermes.consumers.consumer.oauth.OAuthSubscriptionAccessTokens;
+import pl.allegro.tech.hermes.consumers.consumer.oauth.OAuthSubscriptionHandlerFactory;
+import pl.allegro.tech.hermes.consumers.consumer.oauth.OAuthTokenRequestRateLimiterFactory;
+import pl.allegro.tech.hermes.consumers.consumer.oauth.client.OAuthClient;
+import pl.allegro.tech.hermes.consumers.consumer.oauth.client.OAuthHttpClient;
+import pl.allegro.tech.hermes.domain.oauth.OAuthProviderRepository;
+import pl.allegro.tech.hermes.domain.subscription.SubscriptionRepository;
+import pl.allegro.tech.hermes.infrastructure.zookeeper.ZookeeperPaths;
+
+import javax.inject.Named;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+
+
+@Configuration
+public class OAuthConfiguration {
+
+    @Bean
+    public OAuthTokenRequestRateLimiterFactory oAuthTokenRequestRateLimiterFactory(OAuthProviderRepository oAuthProviderRepository,
+                                                                                   ConfigFactory configFactory) {
+        return new OAuthTokenRequestRateLimiterFactory(oAuthProviderRepository, configFactory);
+    }
+
+    @Bean
+    public OAuthAccessTokens oAuthSubscriptionAccessTokens(OAuthAccessTokensLoader tokenLoader,
+                                                           ConfigFactory configFactory) {
+        return new OAuthSubscriptionAccessTokens(tokenLoader, configFactory);
+    }
+
+    @Bean(initMethod = "start", destroyMethod = "stop")
+    public OAuthClient oAuthHttpClient(@Named("oauth-http-client") HttpClient httpClient,
+                                       ObjectMapper objectMapper) {
+        return new OAuthHttpClient(httpClient, objectMapper);
+    }
+
+    @Bean
+    public OAuthSubscriptionHandlerFactory oAuthSubscriptionHandlerFactory(SubscriptionRepository subscriptionRepository,
+                                                                           OAuthAccessTokens accessTokens,
+                                                                           OAuthTokenRequestRateLimiterFactory rateLimiterLoader) {
+        return new OAuthSubscriptionHandlerFactory(subscriptionRepository, accessTokens, rateLimiterLoader);
+    }
+
+    @Bean
+    public OAuthAccessTokensLoader oAuthAccessTokensLoader(SubscriptionRepository subscriptionRepository,
+                                                           OAuthProviderRepository oAuthProviderRepository,
+                                                           OAuthClient oAuthClient,
+                                                           HermesMetrics metrics) {
+        return new OAuthAccessTokensLoader(subscriptionRepository, oAuthProviderRepository, oAuthClient, metrics);
+    }
+
+    @Bean
+    public OAuthProvidersNotifyingCache oAuthProvidersNotifyingCache(@Named(CuratorType.HERMES) CuratorFramework curator,
```

Comment:
Because some of them are from the `hermes-common` module and are used by `hermes-frontend` as well. See: [CommonBinder](https://github.com/allegro/hermes/blob/master/hermes-common/src/main/java/pl/allegro/tech/hermes/common/di/CommonBinder.java#L82)